### PR TITLE
Precursor for the linux backend

### PR DESCRIPTION
**Pull request by harishkrupo**

### CODE REVIEW

**Comment by kalyankondapally** [comment]
You can keep travis, makefile, package config file changes, minigbm guards separately and squash rest as this is the first time we are introducing this.

File: os/linux/weston_frontend.cpp [hunk 1]
```diff
@@ -46,21 +53,160 @@ iahwc_function_ptr_t IAHWC::HookGetFunctionPtr(iahwc_device_t* /* device */,
...
 }
 
-int IAHWC::getnumDisplays() {
-  return 1;
+// private function implementations
+
+int IAHWC::GetNumDisplays(int* num_displays) {
+
+  if (primary_display)
+    *num_displays = 1;
```
Comment (kalyankondapally) [inline os/linux/weston_frontend.cpp]: This can be always one.

File: os/linux/weston_frontend.cpp [hunk 2]
```diff
@@ -129,100 +133,126 @@ int IAHWC::GetDisplayInfo(uint32_t config, int attribute, int32_t* value) {
...
+int IAHWC::IAHWCLayer::SetAcquireFence(int32_t acquire_fence) {
+
+  iahwc_layer_->SetAcquireFence(acquire_fence);
+
+  return IAHWC_ERROR_NONE;
+
+}
+
+hwcomposer::HwcLayer* IAHWC::IAHWCLayer::GetLayer() {
+
```
Comment (kalyankondapally) [inline os/linux/weston_frontend.cpp]: extra white line

File: os/linux/weston_frontend.cpp [hunk 3]
```diff
@@ -129,100 +133,126 @@ int IAHWC::GetDisplayInfo(uint32_t config, int attribute, int32_t* value) {
...
-  hwc_layer->SetNativeHandle(hwc_handle);
-  hwc_layer->SetAcquireFence(acquire_fence);
+  iahwc_layer_->SetNativeHandle(hwc_handle_);
 
   return IAHWC_ERROR_NONE;
 
 }
 
+int IAHWC::IAHWCLayer::SetAcquireFence(int32_t acquire_fence) {
+
```
Comment (kalyankondapally) [inline os/linux/weston_frontend.cpp]: extra white lines

File: os/linux/weston_frontend.cpp [hunk 4]
```diff
@@ -129,100 +133,126 @@ int IAHWC::GetDisplayInfo(uint32_t config, int attribute, int32_t* value) {
...
+  iahwc_layer_->SetAcquireFence(acquire_fence);
+
```
Comment (kalyankondapally) [inline os/linux/weston_frontend.cpp]: extra white lines

**Comment by harishkrupo** [comment]
@kalyankondapally This PR now contains only those 4 commits. Will make a new PR for the squashed commit.

**Comment by kalyankondapally** [comment]
merged thanks

**Pull request closed without merging**